widget/embedded/src/components/CustomDestination/CustomDestination.tsx

```diff
@@ -0,0 +1,161 @@
+import type { PropTypes } from './CustomDestination.types';
+import type { MouseEvent } from 'react';
+
+import { i18n } from '@lingui/core';
+import {
+  Alert,
+  ChevronDownIcon,
+  CloseIcon,
+  Divider,
+  IconButton,
+  PasteIcon,
+  Typography,
+  WalletIcon,
+} from '@rango-dev/ui';
+import React, { useEffect, useRef, useState } from 'react';
+
+import { useAppStore } from '../../store/AppStore';
+import { useQuoteStore } from '../../store/quote';
+import { getBlockchainDisplayNameFor } from '../../utils/meta';
+import { isValidAddress } from '../ConfirmWalletsModal/ConfirmWallets.helpers';
+import { CustomCollapsible } from '../CustomCollapsible/CustomCollapsible';
+import { ExpandedIcon } from '../CustomCollapsible/CustomCollapsible.styles';
+
+import {
+  Container,
+  CustomDestinationButton,
+  StyledTextField,
+} from './CustomDestination.styles';
+
+export function CustomDestination(props: PropTypes) {
+  const {
+    blockchain,
+    handleCollapsibleOpenChange,
+    lastStepToBlockchain,
+    open,
+  } = props;
+  const { customDestination, setCustomDestination } = useQuoteStore();
+  const { config } = useAppStore();
+  const blockchains = useAppStore().blockchains();
+  const blockchainName = getBlockchainDisplayNameFor(blockchain, blockchains);
+  const [value, setValue] = useState('');
```

Comment:
I think `value` and `customDestination` are identical. please check if you can remove value and just use `customDestination` to have a single source of truth.

Reply:
I had added that when the custom destination was closed, it wouldn't lose its data, but it created a bug and I removed it.

Reply:
Do you have an opinion on this?

Reply:
I didn't find sth like that in task description. Maybe it's not necessary.